pkg/dashboard/store/metrics/metrics.go

```diff
@@ -146,14 +146,14 @@ func (collector *Collector) collectArchivedWorkflows() {
 
 type Params struct {
 	fx.In
-
+	log             logr.Logger
```

Comment:
```suggestion
	Log             logr.Logger
```
It should use exported field with fx dependency injection.

rest LGTM! You are almost there.